### PR TITLE
Fix auto update docs CI

### PR DESCRIPTION
**Pull request by justinchuby**
### Motivation and Context

The CI pipeline (https://github.com/onnx/onnx/actions/runs/7405970505/job/20149686235?pr=5826) fails with exit status 128. This PR tries to configure the git username to fix the issue.

### CODE REVIEW

**Comment by justinchuby** [comment]
maybe not this. could be authentication error

File: .github/workflows/auto_update_doc.yml [hunk 1]
```diff
@@ -16,11 +16,6 @@ jobs:
 
     steps:
     - uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
-      with:
-        repository: ${{ github.event.pull_request.head.repo.full_name }}
```
Comment (justinchuby) [inline .github/workflows/auto_update_doc.yml:20]: @jcwchen could you help take a look at this PR? Hopefully the changes are fine? Thanks a lot!

File: onnx/backend/test/case/node/__init__.py [hunk 2]
```diff
@@ -391,24 +391,15 @@ def collect_diff_testcases() -> List[TestCase]:
...
-    )
     # obtain list of added or modified files in this PR
-    obtain_diff = subprocess.Popen(
-        ["git", "diff", "--name-only", "--diff-filter=AM", "origin/main", "HEAD"],
+    obtain_diff = subprocess.run(
+        ["git", "diff", "--name-only", "--diff-filter=AM", "origin/HEAD", "HEAD"],
         cwd=cwd_path,
-        stdout=subprocess.PIPE,
-        stderr=subprocess.PIPE,
+        capture_output=True,
```
Comment (justinchuby) [inline onnx/backend/test/case/node/__init__.py:400]:
```suggestion
        capture_output=True,
        check=True,
```

**Comment by justinchuby** [comment]
hmm still the same error

**Review by jcwchen — COMMENTED** [review]
So the error happened from `git diff --name-only --diff-filter=AM origin/HEAD HEAD`? Perhaps try to run the command directly in the CI and see whether there is a precise error message. I tried that command locally on my Windows and the diff result seems fine. If this command still fails unexpectedly, we can also look for other command to get the diff files.

**Comment by justinchuby** [comment]
Looks like a temporary bug.

**Pull request closed without merging**